### DISCUSSION TITLE
Libraries and runtime updates in .NET 9 Preview 1

### DISCUSSION THREAD

**Discussion by richlander** [dotnet/runtime#98372] [Announcements]
.NET 9 Preview 1 includes several new libraries and runtime features. We focused on the following areas:

Libraries:
- System.Collections
- System.Linq
- System.Reflection
- System.Security.Cryptography
- System.Text.Json

Runtime:
- Native AOT
- JIT

Release notes:
* [Libraries](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview1/libraries.md) 
* [Runtime](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview1/runtime.md) 
* [What's new in the .NET in .NET 9](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/overview) documentation.
* [Breaking Changes in .NET 9](https://learn.microsoft.com/dotnet/core/compatibility/9.0)

.NET 9 Preview 1:
* [Discussion](https://aka.ms/dotnet/9/preview1)
* [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview1/README.md)

**Comment by NKnusperer** [comment]
Is there a summary of what has changed regarding RISC-V support?
There is no mention of this in the release notes.
Looking at the merged PRs [1], there is a lot of activity, but what of it is actually part of this release?
[1] https://github.com/dotnet/runtime/pulls?q=is%3Apr+is%3Aclosed+label%3Aarch-riscv+

**Reply by hez2010** [reply]
See the tracking issue: https://github.com/dotnet/runtime/issues/84834
No binaries related to RISC-V will be shipped until the port is completed.

**Reply by MichalPetryka** [reply]
Worth noting that as of now, RISC-V is only a community supported (which in case of CoreCLR means Samsung) platform, not an officially supported one.

**Reply by richlander** [reply]
Right. That's it. We're glad to see that RISC-V effort and are investing a fair bit into getting those changed merged, however, we don't have a story to tell on it. It's both not our story and the extent of our partnership on that is what you see on GitHub.

**Reply by NKnusperer** [reply]
Thank you for the clarification.

**Comment by Symbai** [comment]
Devblog was really great place for watching .NET updates / release notes. Here on Github its really odd. I see a post about "libraries and runtime updates". I click on it, I don't see any release notes. So I clicked on ".NET 9 Preview 1: Release Notes" link. Unfortunately i get to another github page which does NOT show me any release notes. But there is a "Release notes" link too so I clicked on that. I get to another Github discussion post which does NOT show me any release notes. 

Super frustrating. So I went back to this post and clicked on the other "Release notes" link. Another Github page which does NOT show me what has changed. I gave up.

On Devblog we always had a single blog post covering all the relevant information. Why has this changed? I don't mind its also getting posted here but at least bring back the posts on Devblog.

**Reply by richlander** [reply]
![image](https://github.com/dotnet/runtime/assets/2608468/c39218f8-3950-4410-9ed5-c73ffa506609)

Does that help?

We haven't yet nailed this format. I'm in agreement on that.

**Reply by Symbai** [reply]
Thanks, that helped. My mistake was focusing on the name of the links. I'd still prefer a single page that shows me what's new, what's changed, and if there are any breaking changes - a good summary. I'm not in favor of having multiple tabs open, and I can't think of anything else right now where the changelog is split up.
